week-4/E-arrays-of-objects/exercise-2.js

```diff
@@ -7,50 +7,84 @@ Each destination has a name, a distance from Glasgow, and a list of transportati
 3) Print in the console all the destination names more than 300 kms far away and reachable by train.
 */
 
-
 var destination1 = {
-    destinationName: "Edinburgh",
-    distanceKms: 80,
-    transportations: ["car", "bus", "train"]
+  destinationName: "Edinburgh",
+  distanceKms: 80,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination2 = {
-    destinationName: "London",
-    distanceKms: 650,
-    transportations: ["car", "bus", "train"]
+  destinationName: "London",
+  distanceKms: 650,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination3 = {
-    destinationName: "Paris",
-    distanceKms: 900,
-    transportations: ["train", "plane"]
+  destinationName: "Paris",
+  distanceKms: 900,
+  transportations: ["train", "plane"]
 };
 
 var destination4 = {
-    destinationName: "Dublin",
-    distanceKms: 350,
-    transportations: ["plane", "ferry"]
+  destinationName: "Dublin",
+  distanceKms: 350,
+  transportations: ["plane", "ferry"]
 };
 
-var travelDestinations = [destination1, destination2, destination3, destination4];
+var travelDestinations = [
+  destination1,
+  destination2,
+  destination3,
+  destination4
+];
 
 /*
 DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
+function distanceWithin500kms(distance) {
+  return distance.distanceKms < 500;
+}
 
+var destinationNamesWithin500Kms = travelDestinations
+  .filter(distanceWithin500kms)
+  .map(function CityNames(city) {
+    return city.destinationName;
+  }); // Complete here
 
-var destinationNamesWithin500Kms = // Complete here
-
-var destinationNameReachableByFerry = // Complete here
-
-var destinationNamesMoreThan300KmsAwayByTrain = // Complete here (PRINT THE RESULT IN THE CONSOLE USING FOREACH)
+function destinationByFerry(ferry) {
+  return ferry.transportations.includes("ferry");
+}
+var destinationNameReachableByFerry = travelDestinations
+  .filter(destinationByFerry)
+  .map(function cityFerry(city) {
+    return city.destinationName;
+  }); // Complete here
+function destinationByTrain(distance) {
+  return (
+    distance.transportations.includes("train") && distance.distanceKms > 300
```

Comment:
This is good :+1: . But why name the parameter here `distance` ? What does it actually represent? Similarly for `ferry` above. The function is not just checking whether the destination can be reached by train, so its name is also slightly misleading.

Also try to leave some blank space between your function and variable declarations to make the code easier to read.